src/components/ec-radio-btn/ec-radio-btn.spec.ts

```diff
@@ -0,0 +1,301 @@
+    it('@update:modelValue - should be emitted when input is clicked', async () => {
+      const wrapper = mountRadioBtn({
+        options: [
+          { value: 'y', label: 'Yes' },
+          { value: 'n', label: 'No' },
+        ],
+      });
+
+      await wrapper.findByDataTest('ec-radio-btn-0').trigger('click');
+      expect(wrapper.emitted('update:modelValue')?.length).toBe(1);
+    });
+  });
+
+  describe('attrs', () => {
+    it('should render with custom attributes', () => {
+      const wrapper = mountRadioBtn({}, {
+        attrs: {
+          'data-test': 'my-data-test',
+          class: 'my-class',
+          id: 'test-id',
+        },
+      });
+      expect(wrapper.element).toMatchSnapshot();
+    });
+  });
+
+  describe('v-model', () => {
+    it('should render the radio button and toggle v-model value when input is clicked', async () => {
+      const wrapper = mountRadioBtn({
+        label: 'Test label prop',
+        options: [
+          { value: 'y', label: 'Yes' },
+          { value: 'n', label: 'No' },
+        ],
+        modelValue: 'y',
+      });
+
+      await wrapper.findByDataTest('ec-radio-btn-1').trigger('click');
+      expect(wrapper.emitted()['update:modelValue'][0]).toEqual(['n']);
+    });
+
+    it('should not change the value of v-model if disabled is enabled and user clicks the radio button', async () => {
+      const wrapper = mountRadioBtn({
+        label: 'Test label prop',
+        options: [
+          { value: 'y', label: 'Yes' },
+          { value: 'n', label: 'No' },
+        ],
+        modelValue: 'y',
+        isDisabled: true,
+      });
+
+      await wrapper.findByDataTest('ec-radio-btn-1').trigger('click');
+      expect(wrapper.emitted()['update:modelValue']).toEqual(undefined);
+    });
+  });
+
+  describe('focus', () => {
+    it('should be focused', async () => {
+      const elem = document.createElement('div');
+      document.body.appendChild(elem);
+      const wrapper = mountRadioBtn(
+        {
+          options: [
+            { value: 'y', label: 'Yes' },
+            { value: 'n', label: 'No' },
+          ],
+        },
+        {
+          attachTo: elem,
+        },
+      );
+
+      await wrapper.findByDataTest('ec-radio-btn__input-0').trigger('focus');
+      expect(wrapper.findByDataTest('ec-radio-btn__radio-icon-wrapper-0').element).toMatchSnapshot();
+    });
+
+    it('should be checked and focused', async () => {
+      const elem = document.createElement('div');
+      document.body.appendChild(elem);
+      const wrapper = mountRadioBtn(
+        {
+          options: [
+            { value: 'y', label: 'Yes' },
+            { value: 'n', label: 'No' },
+          ],
+          modelValue: 'y',
+        },
+        {
+          attachTo: elem,
+        },
+      );
+
+      await wrapper.findByDataTest('ec-radio-btn__input-0').trigger('focus');
+      expect(wrapper.findByDataTest('ec-radio-btn__radio-icon-wrapper-0').element).toMatchSnapshot();
```

Comment:
can we assert here with `document.activeElement` and check it it is checked not using snapshot?